x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/OptimizerRules.java

```diff
@@ -319,28 +317,23 @@ public Expression rule(BinaryLogic e) {
 
         // combine conjunction
         private static Expression propagate(And and) {
-            List<Range> ranges = new ArrayList<>();
             // Only equalities, not-equalities and inequalities with a foldable .right are extracted separately;
             // the others go into the general 'exps'.
-            // TODO: In 105217, this should change to EsqlBinaryComparison, but it doesn't exist in this branch yet
-            List<BinaryComparison> equals = new ArrayList<>();
+            List<EsqlBinaryComparison> equals = new ArrayList<>();
             List<NotEquals> notEquals = new ArrayList<>();
-            List<BinaryComparison> inequalities = new ArrayList<>();
+            List<EsqlBinaryComparison> inequalities = new ArrayList<>();
             List<Expression> exps = new ArrayList<>();
 
             boolean changed = false;
 
             for (Expression ex : Predicates.splitAnd(and)) {
-                if (ex instanceof Range) {
```

Comment:
`Range` here refers to a QL Range, i.e. a SQL between statement. ESQL doesn't support those, so we can drop this usage.